app/_components/PoiCard.tsx

```diff
@@ -151,9 +154,8 @@ export function PoiCard({
             onClick={():void  => {
               if (user) {
                 if (!handleCheckUserInSearchZone()) {
-                  if (trackingPinContext) {
-                    trackingPinContext.setTrackingPin(payload);
-                    setShowPopup && setShowPopup(false);
+                  if (importantPinContext) {
+                    importantPinContext.setTrackingPin(payload);
```

Comment:
I mentioned this to @deanachou in another PR, but there is a lot of nested `if` statements here, and I think it would be best if we reduced them.